### PR TITLE
Add config option to specify line_ending_export_style for fixed files

### LINKED ISSUE #797: Feature request: Enable use of unix-style line endings when running sqlfluff fix command on Windows
In our environment, we use Windows PCs for development, but the company standard is that SQL files have unix-style line endings (LF).  We use bash under WSL a lot, and set Windows editors to use LF only.  However, the only python we have available is on Windows. When running sqlfluff fix under python on Windows, the file gets changed from LF to Windows-style line endings CRLF.

I understand that this is python default behavior, but it would be nice to be able to over-ride the default and choose LF only, all the time.

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

This PR fixes #797.
We provide a config option for the line ending to be used when exporting fixed sqlfluff files.
This defaults to the autodetect (lf for linux/mac, crlf for windows) or can be explicitly specified.

Example of getting crlf output on a mac:
Screen-Recording-2021-11-27-at-22 28 19

N.B. this is not a rule, python imports using universal newlines (all treated as \n) so has no impact on parsing/rule evaluation (also it's not like the users can do anything about it themselves if we flag it). This is only applied when the fixed file is written so non-fixed files will be unimpacted (if a more universal conversion to unix-style lines is required then using a program like `dos2unix` in their CI would be recommended).

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by jpy-git** [comment]
FYI investigating extending this fix to all files supplied to the fix command so hold off on review for now

File: src/sqlfluff/core/linter/linter.py [hunk 1]
```diff
@@ -302,6 +303,35 @@ def _warn_unfixable(code: str):
...
 
+    @staticmethod
+    def get_line_ending_export_style(config: FluffConfig) -> Optional[str]:
+        """Get the line_ending_export_style to be used to export the fixed file.
+
+        Options are:
+        - autodetect: Use system default line ending.
+        - lf: Unix-style line ending.
+        - crlf: Windows-style line ending.
+        """
```
Comment (WittierDinosaur) [inline src/sqlfluff/core/linter/linter.py:314]: You might want to add in Mac class style endings, which are just cr

Comment (jpy-git) [reply src/sqlfluff/core/linter/linter.py:314]: @WittierDinosaur good point, it's just `cr: \r` so easy to add 👍

**Comment by jpy-git** [comment]
Closing this PR. Reasoning being that this method although simple will only impact fixed files and could lead to confusion where users think their newlines will be fixed for files with no other linting issues.
I investigated extending this method to apply to all files supplied to fix but without fairly significant changes to how errors are flagged/cli messages are logged this is not possible.
The good news is I think I have a good approach to implement this via a rule which will follow shortly 👍

**Pull request closed without merging**